### ISSUE TITLE
ENH: Add `lsuffix=`/`rsuffix=` kwargs to `pd.merge()` and `df.merge()`

### ISSUE THREAD

**Issue by wany-oh** [pandas-dev/pandas#46046]
It's weird that it's in `join()` but not in `merge()`, and the current behavior (passing the length-2 list-like) of `suffixes=` kwarg is somewhat confusing syntactically.

expected:

```python
>>> df1 = pd.DataFrame({"lkey": ["foo", "bar", "baz", "foo"], "value": [1, 2, 3, 5]})
>>> df2 = pd.DataFrame({"rkey": ["foo", "bar", "baz", "foo"], "value": [5, 6, 7, 8]})

>>> df1.merge(df2, left_on="lkey", right_on="rkey", lsuffix="_left", rsuffix="_right")
  lkey  value_left rkey  value_right
0  foo           1  foo            5
1  foo           1  foo            8
2  foo           5  foo            5
3  foo           5  foo            8
4  bar           2  bar            6
5  baz           3  baz            7
```

**Comment by rhshadrach** [comment]
Agreed it's not ideal that the syntax is different between the two, but I'm fairly negative on going through a deprecation on this, at least at this time. There are any argument outside of personal opinion as to which syntax is better?

**Comment by jreback** [comment]
pd.merge is the more general case (and likely more used than join)

so i think we should unify these
args

+1 on deprecation of lsuffix / rsuffix
in favor of suffixes

**Comment by wany-oh** [comment]
> Agreed it's not ideal that the syntax is different between the two, but I'm fairly negative on going through a deprecation on this, at least at this time. There are any argument outside of personal opinion as to which syntax is better?

As you said, it depends on personal preference as to which is better.
So, on the contrary, some people may say that the `suffixes=` argument should be added to `join()`.

In addition, I am not saying that the current argument should be deprecated.
